File: app/src/main/java/com/boardgamegeek/provider/BuddiesIdAvatarProvider.java
package com.boardgamegeek.provider;

import android.net.Uri;

import com.boardgamegeek.provider.BggContract.Buddies;

public class BuddiesIdAvatarProvider extends IndirectFileProvider {
	@Override
	protected String getPath() {
		return "buddies/*/avatars";
	}

	@Override
	protected Uri getFileUri(Uri uri) {
		return Buddies.buildBuddyUri(Buddies.getBuddyName(uri));
	}

	@Override
	protected String getColumnName() {
		return Buddies.AVATAR_URL;
	}

	@Override
	protected String getContentPath() {
		return BggContract.PATH_AVATARS;
	}
}

File: app/src/main/java/com/boardgamegeek/provider/BuddiesIdAvatarProvider.kt
package com.boardgamegeek.provider

import android.net.Uri
import com.boardgamegeek.provider.BggContract.*
import com.boardgamegeek.provider.BggContract.Companion.PATH_AVATARS
import com.boardgamegeek.provider.BggContract.Companion.PATH_BUDDIES

class BuddiesIdAvatarProvider : IndirectFileProvider() {
    override val path = "$PATH_BUDDIES/*/$PATH_AVATARS"

    override val contentPath = PATH_AVATARS

    override val columnName = Buddies.Columns.AVATAR_URL

    override fun getFileUri(uri: Uri) = Buddies.buildBuddyUri(Buddies.getBuddyName(uri))
}
